packages/scene-composer/src/utils/entityModelUtils/sceneLayerUtils.ts

```diff
@@ -0,0 +1,77 @@
+import { CreateEntityCommandInput, PropertyRequest, UpdateEntityCommandInput } from '@aws-sdk/client-iottwinmaker';
+
+import {
+  DEFAULT_LAYER_COMPONENT_NAME,
+  DEFAULT_LAYER_RELATIONSHIP_NAME,
+  DEFAULT_NODE_COMPONENT_NAME,
+  LAYER_COMPONENT_TYPE_ID,
+  LAYER_ROOT_ENTITY_ID,
+  LayerType,
+  layerPropertyNames,
+} from '../../common/entityModelConstants';
+import { getGlobalSettings } from '../../common/GlobalSettings';
+import { generateUUID } from '../mathUtils';
+
+export const createLayerId = (layerName: string) => {
+  return `LAYER_${layerName}_${generateUUID()}`;
+};
+
+export const createLayer = (layerName: string, type: LayerType) => {
+  const layerId = createLayerId(layerName);
+  const input: CreateEntityCommandInput = {
+    workspaceId: undefined,
+    entityId: layerId,
+    parentEntityId: LAYER_ROOT_ENTITY_ID,
+    entityName: layerId,
+    components: {
+      [DEFAULT_LAYER_COMPONENT_NAME]: {
+        componentTypeId: LAYER_COMPONENT_TYPE_ID,
+        properties: {
+          [layerPropertyNames.type]: {
+            value: {
+              stringValue: type,
+            },
+          },
+        },
+      },
+    },
+  };
+
+  if (type === LayerType.Relationship) {
+    input.components![DEFAULT_LAYER_COMPONENT_NAME].properties![layerPropertyNames.relationshipName] = {
```

Comment:
What's the reason for using exclamation mark?

Reply:
the type of `components` in `CreateEntityCommandInput` can be undefined and typescript is not smart enough to know that I've assigned value to it above